src/java.desktop/unix/native/common/awt/CUPSfuncs.c

```diff
@@ -219,6 +214,53 @@ Java_sun_print_CUPSPrinter_getCupsDefaultPrinter(JNIEnv *env,
     return cDefPrinter;
 }
 
+/*
+ * Returns list of default local printers
+ */
+JNIEXPORT jobjectArray JNICALL
+Java_sun_print_CUPSPrinter_getCupsDefaultPrinters(JNIEnv *env,
+                                                        jobject printObj)
+{
+    cups_dest_t *dests;
+    int i, num_dests;
+    jstring utf_str;
+    jclass cls;
+    jobjectArray nameArray = NULL;
+
+    cls = (*env)->FindClass(env, "java/lang/String");
+    CHECK_NULL_RETURN(cls, NULL);
+
+    num_dests = j2d_cupsGetDests(&dests);
+
+    if (dests == NULL) {
+        return NULL;
+    }
+
+    nameArray = (*env)->NewObjectArray(env, num_dests, cls, NULL);
+    if (nameArray == NULL) {
+        j2d_cupsFreeDests(num_dests, dests);
+        DPRINTF("CUPSfuncs::bad alloc new array\n", "")
+        (*env)->ExceptionClear(env);
+        JNU_ThrowOutOfMemoryError(env, "OutOfMemoryError");
```

Comment:
I find this weird. What is the ExceptionClear for ? The only possible exception here is for
an OOME which might be thrown by NewObjectArray. So you clear that and then re-create it ?
And who do will catch this ? What's the point ? Maybe just clear and return null.

Reply:
I can't agree with the apparent reasoning that "well that thing over there is wrong, so it's ok for me to add something here that's wrong too".

And the getMedia(..) case is different.  It doesn't re-create the same exception.
It wants to create an OOME which is not actually thrown by GetStringUTFChars() since it needs to throw some exception and with an appropriate string reason.
And the clear there is just prudence since it is creating a new one and can't be 100% sure there isn't one pending.

In your case the JNI spec documents OOME 
https://docs.oracle.com/en/java/javase/17/docs/specs/jni/functions.html#newobjectarray

So in your case it is pointless. Please just remote the Clear()

Reply:
ExceptionClear(env) is removed from the `if (nameArray == NULL)` check.

Reply:
I should have said (as I did in the beginning) remove the clear - and the new throw - and just return null.

Reply:
JNU_ThrowOutOfMemoryError is removed from code where nameArray and utf_str are checked to NULL.